recipe/meta.yaml

```diff
@@ -19,7 +19,8 @@ requirements:
     - cross-python_{{ target_platform }}  # [build_platform != target_platform]
     - crossenv                            # [build_platform != target_platform]
     - {{ compiler('c') }}
-    - {{ compiler('rust') }}
+    #- {{ compiler('rust') }}
```

Comment:
You need this here as well

Reply:
It worked much better without it:
```
requirements:
  build:
    - python                              # [build_platform != target_platform]
    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
    - crossenv                            # [build_platform != target_platform]
    - maturin >=0.13,<0.14                # [build_platform != target_platform]
    - {{ compiler('c') }}
    - {{ compiler('rust') }}
    - rust 1.65.0.dev20220819
    - posix                               # [win]
    - cmake
    - make                                # [unix]
    - cargo-bundle-licenses
```

```
...
Mamba failed to solve:
 - rust 1.65.0.dev20220819
 - cmake
 - make
 - gcc_linux-64 10.*
 - rust_linux-64
 - cargo-bundle-licenses

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package rust_linux-64-1.56.0.dev20210816-h383556e_0 requires rust 1.56.0.dev20210816.*, but none of the providers can be installed
- 
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 124, in mamba_get_install_actions
    solution = solver.solve_for_action(_specs, prefix)
  File "/opt/conda/lib/python3.9/site-packages/boa/core/solver.py", line 226, in solve_for_action
    t = self.solve(specs)
  File "/opt/conda/lib/python3.9/site-packages/boa/core/solver.py", line 216, in solve
    raise RuntimeError("Solver could not find solution." + error_string)
RuntimeError: Solver could not find solution.Mamba failed to solve:
 - rust 1.65.0.dev20220819
 - cmake
 - make
 - gcc_linux-64 10.*
 - rust_linux-64
 - cargo-bundle-licenses

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package rust_linux-64-1.56.0.dev20210816-h383556e_0 requires rust 1.56.0.dev20210816.*, but none of the providers can be installed
- 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/conda-mambabuild", line 10, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 239, in main
    call_conda_build(action, config)
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 211, in call_conda_build
    result = api.build(
  File "/opt/conda/lib/python3.9/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 3091, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 2054, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 788, in expand_outputs
    for (output_dict, m) in deepcopy(_m).get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "/opt/conda/lib/python3.9/site-packages/conda_build/metadata.py", line 2121, in get_output_metadata_set
    conda_packages = finalize_outputs_pass(ref_metadata, conda_packages, pass_no=0,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/metadata.py", line 782, in finalize_outputs_pass
    fm = finalize_metadata(om, parent_metadata=parent_metadata,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 546, in finalize_metadata
    build_unsat, host_unsat = add_upstream_pins(m,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 386, in add_upstream_pins
    build_deps, build_unsat, extra_run_specs_from_build = _read_upstream_pin_files(m, 'build',
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 373, in _read_upstream_pin_files
    deps, actions, unsat = get_env_dependencies(m, env, m.config.variant,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 138, in get_env_dependencies
    actions = environ.get_install_actions(tmpdir, tuple(dependencies), env,
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 126, in mamba_get_install_actions
    conflict_packages = parse_problems(str(e))
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 41, in parse_problems
    assert words[1] == "package", f"words = {repr(words)}"
AssertionError: words = ['-', '-', 'package', 'rust_linux-64-1.56.0.dev20210816-h383556e_0', 'requires', 'rust', '1.56.0.dev20210816.*,', 'but', 'none', 'of', 'the', 'providers', 'can', 'be', 'installed']
Traceback (most recent call last):
  File "./build-locally.py", line 105, in <module>
    main()
  File "./build-locally.py", line 93, in main
    run_docker_build(ns)
  File "./build-locally.py", line 29, in run_docker_build
    subprocess.check_call([script])
  File "/home/ghuls/software/anaconda3/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 1.

```

Reply:
Without it cross compilation would not work. To fix that error, we need to update https://github.com/conda-forge/rust-activation-feedstock as well for 1.65.0

Reply:
Can  you make a nightly for the rust-activation-feedstock?

Reply:
Can you send a PR?

Reply:
It doesn't have an example for making it for a nightly version as far as I can see.